webclient/grpc/src/main/java/io/helidon/webclient/grpc/ClientUriSuppliers.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.webclient.grpc;
+
+import java.net.URI;
+import java.security.SecureRandom;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import io.helidon.webclient.api.ClientUri;
+
+/**
+ * Defines the {@code ClientUriSupplier} interface and provides a few basic
+ * implementations of it.
+ */
+public class ClientUriSuppliers {
+
+    /**
+     * Supplies an iterator that returns URIs chosen in order from
+     * first to last.
+     */
+    public static class OrderedSupplier implements ClientUriSupplier {
+
+        private final Iterator<ClientUri> clientUris;
+
+        /**
+         * Creates an ordered supplier.
+         *
+         * @param clientUris array of client URIs
+         * @return new supplier
+         */
+        public static OrderedSupplier create(ClientUri... clientUris) {
+            return new OrderedSupplier(List.of(clientUris));
+        }
+
+        /**
+         * Creates an ordered supplier.
+         *
+         * @param clientUris collection of client URIs
+         * @return new supplier
+         */
+        public static OrderedSupplier create(Collection<ClientUri> clientUris) {
+            return new OrderedSupplier(clientUris);
```

Comment:
The `RoundRobinSupplier` copies the collection passed into its `create` method (or rather it turns it into an array). In this method, the `OrderedSupplier` will create an iterator directly from the collection parameter. If this collection ever changes that will possibly cause a concurrent modification exception.

Reply:
Fixed